### DISCUSSION TITLE
How to output into another format? eg: sparql+xml

### DISCUSSION THREAD

**Discussion by folkvir** [comunica/comunica#575] [Q&A] [answered]
Do you have any documentation on how to output my result into sparql-xml?

There is this [package](https://github.com/comunica/comunica/tree/master/packages/actor-sparql-serialize-sparql-xml) but it is not documented.

```js
async function start(server, query) {
  const newEngine = require('@comunica/actor-init-sparql').newEngine;
    newEngine().query(query, {
      sources: [server],
    }).then((result) => {
      result.bindingsStream.on('data', d => {
        console.log(d.toObject())
      })
    }).catch(e => {
        console.error(e)
    })
}
```
Do I have to add something in the options object?

**Comment by rubensworks** [comment, marked as answer]
Your query engine should have a [`.resultToString(queryResult)` method](https://github.com/comunica/comunica/blob/master/packages/actor-init-sparql/lib/ActorInitSparql-browser.ts#L205-L213), that by should be able to serialize to SPARQL XML by passing `application/sparql-results+xml` as media type.

This is similar to the question in #452. (I haven't had the time yet to document this, so I'll leave this issue open)

**Comment by folkvir** [comment]
Thank you for your response :)

This works perfectly:

```js
async function start(server, query) {
  const newEngine = require('@comunica/actor-init-sparql').newEngine;
  const engine = newEngine()
  engine.query(query, {
    sources: [server],
  }).then((result) => {
    engine.resultToString(result, 'application/sparql-results+xml', result.context).then((d) => {
      res = ''
      d.data.on('data', (a) => {
        res += a
      })
      d.data.on('end', () => {
        console.log(res)
      })
    })
  }).catch(e => {
    console.error(e)
  })
}
```

**Comment by rubensworks** [comment]
If you want to retain the streaming nature of results,
you can simplify your code to something like this:

```javascript
async function start(server, query) {
  const newEngine = require('@comunica/actor-init-sparql').newEngine;
  const engine = newEngine()
  engine.query(query, {
    sources: [server],
  }).then((result) => {
    engine.resultToString(result, 'application/sparql-results+xml', result.context).then((d) => {
      d.data.pipe(process.stdout);
    })
  }).catch(e => {
    console.error(e)
  })
}
```

(the explicit .then calls can probably also be simplified to await's)
